src/hotspot/share/services/rsswatch.cpp

```diff
@@ -109,11 +109,16 @@ void RssWatcher::initialize(const char* limit_option) {
   if (parse_percentage(s, &s, &percentage)) {
     is_absolute = false;
   } else {
-    if (!parse_integer(s, (char**)&s, &limit) || limit == 0) {
+    if (!parse_integer(s, (char**)&s, &limit) || (limit == 0)) {
       vm_exit_during_initialization("Failed to parse RssLimit", "Not a valid limit size");
     }
   }
 
+  if (os::get_RSS() == 0) {
+    log_warning(os, rss)("RssLimit specified, but not supported by the Operating System.");
+    return;
+  }
```

Comment:
This seems a strange place for this. I would expect it to be in the init function.

Reply:
Sorry -ignore this. The git UI had me confused about where this was.